db_master_deploy/deploy.sh

```diff
@@ -0,0 +1,379 @@
+            source_objects=$OPTARG
+            IFS=',' read -ra array_source <<< "$source_objects"
+            ;;
+        t)
+            target=$OPTARG
+            ;;
+        a)
+            timestamp=$OPTARG
+            ;;
+        \? )
+            display_usage 1>&5 2>&6
+            exit 1
+            ;;
+        *)
+            display_usage 1>&5 2>&6
+            exit 1
+            ;;
+    esac
+done
+
+check_table() {
+    # check if source table exists, views cannot be deployed
+    if [[ -z `psql -lqt -h localhost -U pgkogis -c "SELECT table_catalog||'.'||table_schema||'.'||table_name FROM information_schema.tables where lower(table_type) not like 'view'" -d $source_db  2> /dev/null | egrep "\b$source_id\b"` ]]; then
+        echo "source table does not exist $source_id " >&2
+        exit 1
+    fi
+
+    # check if target table exists 
+    if [[ -z `psql -lqt -h localhost -U pgkogis -c "SELECT table_catalog||'.'||table_schema||'.'||table_name FROM information_schema.tables where lower(table_type) not like 'view'" -d $target_db  2> /dev/null | egrep "\b$target_id\b"` ]]; then
+        "target table does not exist $target_id." >&2
+        exit 1
+    fi
+
+    # check if source and target table have the same structure (column name and data type)
+    source_columns=`psql -U pgkogis -h localhost -d $source_db -Atc "select column_name,data_type FROM information_schema.columns WHERE table_schema = '$source_schema' AND table_name = '$source_table' order by 1;"`
+    target_columns=`psql -U pgkogis -h localhost -d $target_db -Atc "select column_name,data_type FROM information_schema.columns WHERE table_schema = '$target_schema' AND table_name = '$target_table' order by 1;"`
+    if [ ! "$source_columns" == "$target_columns" ]; then
+        echo "structure of source and target table is different." >&2
+        exit 1
+    fi
+    
+    # get count of referencing tables/constraints, skip table if it is referenced by other tables
+    referencing_tables_sql="
+    SELECT count(1)
+        FROM pg_catalog.pg_constraint r
+        WHERE  r.contype = 'f' 
+        AND conrelid::regclass != confrelid::regclass
+        AND confrelid::regclass = '$source_schema.$source_table'::regclass;
+    "
+    referencing_tables=$(psql -qAt -h localhost -U pgkogis -d $source_db -c "$referencing_tables_sql")
+    if [ $referencing_tables -gt 0 ] 
+    then 
+        echo "cannot copy table $source_id, table is referenced by $referencing_tables objects, use db_copy instead." >&2
+        continue
+    fi
+}
+
+check_database() {
+    # check if source database exists
+    if [[ -z `psql -lqt -h localhost -U pgkogis | egrep "\b$source_db\b" 2> /dev/null` ]]; then
+        echo "No existing databases are named $source_db." >&2
+        exit 1
+    fi
+}
+
+bod_create_archive() {
+    #BOD archiving
+    if [[ ${source_db%_*} == bod ]]
+    then
+        if [[ ${#timestamp} > 0 ]]
+        then
+            if [[ ! $timestamp =~ (^[a-zA-Z0-9]+$)  ]]
```

Comment:
Yes thats punkrock :-)
I have tried to apply some style rules 
https://google-styleguide.googlecode.com/svn/trunk/shell.xml#Comments